src/Language/Fixpoint/Types/Refinements.hs

```diff
@@ -493,7 +494,9 @@ instance Fixpoint Expr where
 
   simplify (PAnd ps)
     | any isContraPred ps = PFalse
-    | otherwise           = PAnd $ filter (not . isTautoPred) $ map simplify ps
+    | otherwise           = PAnd $ dedup . flattenRefas . filter (not . isTautoPred) $ map simplify ps
```

Comment:
```suggestion
                                    -- Note: Performance of some tests is very sensitive to this code. See #480 
    | otherwise           = PAnd $ dedup . flattenRefas . filter (not . isTautoPred) $ map simplify ps
```